File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    answer = ''
    dic = defaultdict(int)
    
    for i in participant:
        dic[i] += 1
    
    for j in completion:
        dic[j] -= 1
        
    for k, v in dic.items():
        if v == 1:
            answer = k
        
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> dic = new HashMap<>();
        
        for (String i : participant) {
            dic.put(i, dic.getOrDefault(i, 0)+1);
        }
        
        for (String j : completion) {
            dic.put(j, dic.get(j) - 1);
        }
        
        for (String key : dic.keySet()) {
            if (dic.get(key) != 0){
                answer = key;
            }
        }
        
        return answer;
    }
}